pkg/controller/certificate-shim/helper.go

```diff
@@ -268,5 +269,26 @@ func translateAnnotations(crt *cmapi.Certificate, ingLikeAnnotations map[string]
 		}
 	}
 
+	if customAnnotationsRegexString, found := ingLikeAnnotations[cmapi.IngressSecretTemplateAnnotations]; found {
+		customAnnotationsRegex, err := regexp.Compile(customAnnotationsRegexString)
+		if err != nil {
+			return fmt.Errorf("%w %q: error parsing regexp: %q", errInvalidIngressAnnotation, cmapi.IngressSecretTemplateAnnotations, customAnnotationsRegexString)
+		}
+		for annotationKey, annotationValue := range ingLikeAnnotations {
+			match := customAnnotationsRegex.FindString(annotationKey)
+			if len(match) == len(annotationKey) {
+				if strings.HasPrefix(annotationKey, "cert-manager.io/") {
+					return fmt.Errorf("%w %q: regex must not match cert-manager.io/ annotations: %q", errInvalidIngressAnnotation, cmapi.IngressSecretTemplateAnnotations, customAnnotationsRegexString)
```

Comment:
Could you add a `'` or \` around `cert-manager.io/`?

Reply:
Not needed anymore since I've updated the PR to use a different approach now!